chemreg/compound/serializers.py

```diff
@@ -172,7 +186,36 @@ class CompoundSerializer(PolymorphicModelSerializer):
     """The serializer for both ill-defined and defined compounds."""
 
     polymorphic_serializers = [DefinedCompoundSerializer, IllDefinedCompoundSerializer]
+    serializer_kwargs = {
+        DefinedCompoundSerializer: ["override", "is_admin"],
+        IllDefinedCompoundSerializer: ["is_admin"],
+    }
 
     class Meta:
         model = BaseCompound
-        fields = ["cid"]
+
+
+class CompoundDeleteSerializer(serializers.Serializer):
```

Comment:
do we need to validate that the `replacement_cid` comes from the same child class, ie., Defined vs. IllDefined, right now I can replace an IllDefinedCompound with a DefinedCompound cid. Here's a clip of what migh go into a `validate` method here...
```
    def validate_replacement_cid(self, value):
        Compound = self.instance._meta.model
        if not Compound.objects.filter(
            cid=self.initial_data["replacement_cid"]
        ).exists():
            raise ValidationError("Go to jail!")
        return value
```

not sure if this would be the best way to do it, or if it even HAS to be done, but it's a thought.

Reply:
this doesn't need to be validated from convo [here](https://github.com/chemical-curation/chemcurator_django/issues/39#issuecomment-641436101)